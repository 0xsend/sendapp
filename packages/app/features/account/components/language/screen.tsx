import { Card, Paragraph, Separator, Spinner, XStack, YStack } from '@my/ui'
import { SettingsHeader } from 'app/features/account/components/SettingsHeader'
import { useTranslation } from 'react-i18next'
import { useCallback, useMemo, useState } from 'react'
import { Pressable, Platform } from 'react-native'
import { Check } from '@tamagui/lucide-icons'
import { getSupportedLocaleOptions } from 'app/i18n/locales'
import { useHoverStyles } from 'app/utils/useHoverStyles'

export const LanguagePreferences = () => {
  const { i18n, t } = useTranslation('settings')
  const [pendingLocale, setPendingLocale] = useState<string | null>(null)
  const resolvedLanguage = i18n.resolvedLanguage ?? i18n.language

  const options = useMemo(() => getSupportedLocaleOptions(), [])
  const hoverStyles = useHoverStyles()

  const handleSelect = useCallback(
    async (locale: string) => {
      if (locale === resolvedLanguage || pendingLocale === locale) {
        return
      }

      setPendingLocale(locale)
      try {
        await i18n.changeLanguage(locale)
      } finally {
        setPendingLocale(null)
      }
    },
    [i18n, pendingLocale, resolvedLanguage]
  )

  return (
    <YStack gap={'$3.5'} w={'100%'}>
      {Platform.OS === 'web' && <SettingsHeader>{t('language.title')}</SettingsHeader>}
      <Card gap={'$3'} padded size={'$4'}>
        <Paragraph size={'$4'} color={'$color12'}>
          {t('language.description')}
        </Paragraph>
        {pendingLocale ? (
          <Paragraph size={'$2'} color={'$lightGrayTextField'}>
            {t('language.updating')}
          </Paragraph>
        ) : null}
        <Separator boc={'$darkGrayTextField'} opacity={0.2} />
        <YStack>
          {options.map((option, index) => {
            const isActive = option.value === resolvedLanguage
            const isPending = pendingLocale === option.value

            return (
              <YStack key={option.value}>
                <Pressable
                  accessibilityRole="button"
                  accessibilityState={{ selected: isActive, busy: isPending }}
                  disabled={isPending}
                  onPress={() => handleSelect(option.value)}
                >
                  <XStack
                    ai="center"
                    jc="space-between"
                    px={'$2'}
                    py={'$3'}
                    br={'$4'}
                    hoverStyle={hoverStyles}
                  >
                    <Paragraph size={'$5'}>{option.autonym}</Paragraph>
                    {isPending ? (
                      <Spinner size="small" />
                    ) : isActive ? (
                      <Check size={16} color={'$primary'} />
                    ) : null}
                  </XStack>
                </Pressable>
                {index < options.length - 1 && (
                  <Separator boc={'$darkGrayTextField'} opacity={0.15} />
                )}
              </YStack>
            )
          })}
        </YStack>
      </Card>
    </YStack>
  )
}
