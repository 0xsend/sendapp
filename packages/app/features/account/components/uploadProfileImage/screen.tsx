import { SizableText, YStack } from '@my/ui'
import { IconRefresh } from 'app/components/icons'
import { useSupabase } from 'app/utils/supabase/useSupabase'
import { useUser } from 'app/utils/useUser'
import { decode } from 'base64-arraybuffer'
import * as ImagePicker from 'expo-image-picker'
import React, { Ref, forwardRef, useImperativeHandle, useState } from 'react'

export interface UploadAvatarRefObject {
  pickImage: () => void
}

export const UploadAvatar = forwardRef(function UploadAvatar(
  { children }: { children: React.ReactNode },
  ref: Ref<UploadAvatarRefObject>
) {
  const { user, updateProfile } = useUser()
  const supabase = useSupabase()
  const [errMsg, setErrMsg] = useState('')

  useImperativeHandle(ref, () => ({ pickImage }))

  const pickImage = async () => {
    // No permissions request is necessary for launching the image library
    const result = await ImagePicker.launchImageLibraryAsync({
      mediaTypes: ImagePicker.MediaTypeOptions.Images,
      allowsEditing: true,
      aspect: [1, 1],
      quality: 1,
      base64: true,
    })

    await uploadImage(result)
  }

  const uploadImage = async (pickerResult: ImagePicker.ImagePickerResult) => {
    if (pickerResult.canceled) {
      return
    }
    if (!user) return
    const image = pickerResult.assets[0]
    if (!image) {
      setErrMsg('No image provided.')
      return
    }

    const base64Image = image.base64

    if (!base64Image) {
      setErrMsg('No image provided.')
      return
    }

    const base64Str = base64Image.includes('base64,')
      ? base64Image.substring(base64Image.indexOf('base64,') + 'base64,'.length)
      : base64Image
    const res = decode(base64Str)

    if (!(res.byteLength > 0)) {
      setErrMsg('ArrayBuffer is null')
      // console.error('ArrayBuffer is null')
      return null
    }
    const result = await supabase.storage
      .from('avatars')
      .upload(`${user.id}/${Number(new Date())}.jpeg`, res, {
        contentType: 'image/jpeg',
        upsert: true,
      })
    if (result.error) {
      setErrMsg(result.error.message)
      return
      // console.log(result.error)
      // throw new Error(result.error.message)
    }
    const publicUrlRes = await supabase.storage
      .from('avatars')
      .getPublicUrl(result.data.path.replace('avatars/', ''))

    const { error: update_error } = await supabase
      .from('profiles')
      .update({ avatar_url: publicUrlRes.data.publicUrl })
      .eq('id', user.id)
    if (update_error) {
      return
    }
    await updateProfile()
  }

  return (
    <YStack gap={'$4'}>
      <YStack
        position="relative"
        alignSelf="flex-start"
        flexShrink={1}
        onPress={() => pickImage()}
        cursor="pointer"
      >
        {children}
        <YStack
          position="absolute"
          left={0}
          right={0}
          top={0}
          bottom={0}
          jc="center"
          ai="center"
          zIndex={100}
        >
          <YStack
            backgroundColor="black"
            opacity={0.34}
            borderRadius="$3"
            position="absolute"
            left={0}
            right={0}
            top={0}
            bottom={0}
            $md={{ borderRadius: 12 }}
          />
          <YStack position="absolute" left={0} right={0} top={0} bottom={0} jc="center" ai="center">
            <IconRefresh color="primary" />
          </YStack>
        </YStack>
      </YStack>
      {errMsg ? <SizableText theme="red">{errMsg}</SizableText> : <></>}
    </YStack>
  )
})
