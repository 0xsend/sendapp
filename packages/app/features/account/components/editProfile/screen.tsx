import {
  Button,
  Checkbox,
  FadeCard,
  Paragraph,
  ProfileAvatar,
  Separator,
  Spinner,
  SubmitButton,
  XStack,
  YStack,
} from '@my/ui'
import { SchemaForm } from 'app/utils/SchemaForm'
import { ProfileSchema, useProfileMutation } from 'app/utils/useProfileMutation'
import { useUser } from 'app/utils/useUser'
import { UploadAvatar, type UploadAvatarRefObject } from '../uploadProfileImage/screen'
import { useRef, useState } from 'react'
import type { Tables } from '@my/supabase/database.types'
import { Check } from '@tamagui/lucide-icons'
import { SettingsHeader } from 'app/features/account/settings/components/SettingsHeader'
import { ReadOnlyFieldWithLabel } from 'app/features/account/settings/components/ReadOnlyFieldWithLabel'
import { FieldWithLabel } from 'app/features/account/settings/components/FieldWithLabel'
import { useForm } from 'react-hook-form'
import type { z } from 'zod'

enum FormState {
  Overview = 'Overview',
  ProfileForm = 'ProfileForm',
}

export const EditProfile = () => {
  const [formState, setFormState] = useState<FormState>(FormState.Overview)
  const { profile } = useUser()

  return (
    <YStack w={'100%'}>
      <YStack gap={'$3.5'}>
        <SettingsHeader>Profile</SettingsHeader>
        {(() => {
          switch (true) {
            case !profile:
              return <Spinner size="large" />
            case formState === FormState.Overview:
              return (
                <Overview profile={profile} onPress={() => setFormState(FormState.ProfileForm)} />
              )
            case formState === FormState.ProfileForm:
              return (
                <EditProfileForm
                  profile={profile}
                  onSave={() => setFormState(FormState.Overview)}
                />
              )
          }
        })()}
      </YStack>
    </YStack>
  )
}

const Overview = ({ profile, onPress }: { profile: Tables<'profiles'>; onPress: () => void }) => {
  const { name, about, is_public, avatar_url } = profile
  const avatarRef = useRef<UploadAvatarRefObject>(null)

  return (
    <YStack gap={'$5'}>
      <FadeCard>
        <XStack gap={'$5'} width={'100%'}>
          <UploadAvatar ref={avatarRef}>
            <ProfileAvatar
              avatarUrl={avatar_url ? avatar_url : undefined}
              $gtMd={{ size: 88 }}
              size={88}
            />
          </UploadAvatar>
          <YStack gap={'$2'} ai={'flex-start'}>
            <YStack>
              <Paragraph
                size={'$5'}
                fontWeight={500}
                color={'$lightGrayTextField'}
                $theme-light={{ color: '$darkGrayTextField' }}
              >
                Profile Picture
              </Paragraph>
            </YStack>
            <Button unstyled onPress={() => avatarRef.current?.pickImage()}>
              <Button.Text
                textDecorationLine="underline"
                color="$primary"
                $theme-light={{ color: '$color12' }}
                size={'$5'}
              >
                Update
              </Button.Text>
            </Button>
          </YStack>
        </XStack>
        <FieldWithLabel label={'Name'}>
          <Paragraph size={'$8'} fontWeight={'500'}>
            {name || '-'}
          </Paragraph>
        </FieldWithLabel>
        <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
        <ReadOnlyFieldWithLabel label={'About'} text={about || '-'} />
        <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
        <XStack gap={'$3'} ai={'center'}>
          <Checkbox
            disabled={true}
            checked={!!is_public}
            borderWidth={0}
            backgroundColor={is_public ? '$primary' : '$background'}
            circular={true}
          >
            <Checkbox.Indicator>
              <Check color={'$black'} />
            </Checkbox.Indicator>
          </Checkbox>
          <Paragraph size={'$5'}>{`${is_public ? 'Public' : 'Private'}`} Profile</Paragraph>
        </XStack>
      </FadeCard>
      <SubmitButton theme="green" borderRadius={'$4'} p={'$4'} onPress={onPress}>
        <Button.Text ff={'$mono'} fontWeight={'500'} tt="uppercase" size={'$5'} color={'$black'}>
          edit profile
        </Button.Text>
      </SubmitButton>
    </YStack>
  )
}

function EditProfileForm({ profile, onSave }: { profile: Tables<'profiles'>; onSave: () => void }) {
  const { id, name, about, is_public } = profile
  const { mutateAsync, error } = useProfileMutation(id)
  const form = useForm<z.infer<typeof ProfileSchema>>()

  const handleSubmit = async () => {
    const values = form.getValues()
    await mutateAsync(values)
    onSave()
  }

  return (
    <SchemaForm
      form={form}
      schema={ProfileSchema}
      props={{
        name: {
          'aria-label': 'Name',
          bc: '$color0',
        },
        about: {
          'aria-label': 'Bio',
          placeholder: 'Tell us about yourself',
          backgroundColor: '$color0',
          rows: 2,
          focusStyle: {
            fontStyle: 'normal',
          },
        },
        isPublic: {
          defaultChecked: is_public !== null ? is_public : true,
        },
      }}
      defaultValues={{
        name: name ? name : '',
        about: about ? about : '',
        isPublic: is_public !== null ? is_public : true,
      }}
      formProps={{
        $gtSm: {
          maxWidth: '100%',
        },
      }}
      onSubmit={handleSubmit}
      renderAfter={({ submit }) => (
        <YStack>
          <SubmitButton
            theme="green"
            borderRadius={'$4'}
            p={'$4'}
            mt={'$1'}
            onPress={() => submit()}
          >
            <Button.Text
              ff={'$mono'}
              fontWeight={'500'}
              tt="uppercase"
              size={'$5'}
              color={'$black'}
            >
              SAVE CHANGES
            </Button.Text>
          </SubmitButton>
          {error && (
            <Paragraph marginTop={'$3'} theme="red" color="$color9">
              {error.message}
            </Paragraph>
          )}
        </YStack>
      )}
    >
      {({ name, about, isPublic }) => (
        <FadeCard>
          <FieldWithLabel label={'Name'} gap={'$2'}>
            {name}
          </FieldWithLabel>
          <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
          <FieldWithLabel label={'About'} gap={'$2'}>
            {about}
          </FieldWithLabel>
          <Separator boc={'$silverChalice'} $theme-light={{ boc: '$darkGrayTextField' }} />
          <XStack gap={'$3'} ai={'center'}>
            {isPublic}
            <Paragraph size={'$5'}>Make Profile Public</Paragraph>
          </XStack>
        </FadeCard>
      )}
    </SchemaForm>
  )
}
