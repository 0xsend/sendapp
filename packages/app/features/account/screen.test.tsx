import { expect, test } from '@jest/globals'
import { TamaguiProvider, config } from '@my/ui'
import { render } from '@testing-library/react-native'
import { AccountScreen } from './screen'

jest.mock('app/utils/useUser', () => ({
  useUser: jest.fn().mockReturnValue({
    user: {
      id: '123',
      profile: { referral_code: '123' },
    },
  }),
}))

jest.mock('wagmi', () => ({
  useAccount: jest.fn().mockReturnValue({
    address: '0x123',
    isConnected: false,
  }),
  useConnect: jest.fn().mockReturnValue({
    connectAsync: jest.fn(),
  }),
  useDisconnect: jest.fn().mockReturnValue({
    disconnect: jest.fn(),
  }),
}))

jest.mock('app/utils/getReferralLink', () => ({
  getReferralHref: jest.fn().mockReturnValue('https://send.it/123'),
}))

jest.mock('solito/link', () => ({
  Link: jest.fn(),
}))

jest.mock('app/utils/UseUserReferralsCount', () => ({
  useUserReferralsCount: jest.fn().mockReturnValue(123),
}))

test('AccountScreen', () => {
  const tree = render(
    <TamaguiProvider defaultTheme={'dark'} config={config}>
      <AccountScreen />
    </TamaguiProvider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
