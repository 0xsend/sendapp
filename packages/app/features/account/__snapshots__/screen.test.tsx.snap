// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountScreen renders the account screen: AccountScreen 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "gap": 32,
      "paddingBottom": 32,
      "paddingTop": 32,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
        "zIndex": 4,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "column",
          "gap": 24,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "borderBottomLeftRadius": 7,
            "borderBottomRightRadius": 7,
            "borderTopLeftRadius": 7,
            "borderTopRightRadius": 7,
            "flexDirection": "column",
            "height": 104,
            "justifyContent": "center",
            "maxHeight": 104,
            "maxWidth": 104,
            "minHeight": 104,
            "minWidth": 104,
            "overflow": "hidden",
            "position": "relative",
            "width": 104,
          }
        }
      >
        <View
          style={
            {
              "bottom": 0,
              "flexDirection": "column",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 1,
            }
          }
        >
          <Image
            accessibilityLabel=""
            fullscreen={true}
            onError={[Function]}
            onLoad={[Function]}
            source={
              {
                "uri": "https://example.com",
              }
            }
            style={
              {
                "height": 104,
                "width": 104,
              }
            }
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#3E4A3C",
              "bottom": 0,
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="64"
            bbWidth="64"
            color="#86AE80"
            fill="none"
            focusable={false}
            height={64}
            meetOrSlice={0}
            minX={0}
            minY={0}
            strokeWidth="2"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "color": "#86AE80",
                },
                {
                  "flex": 0,
                  "height": 64,
                  "width": 64,
                },
              ]
            }
            tintColor="#86AE80"
            vbHeight={20}
            vbWidth={20}
            width={64}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "strokeWidth",
                ]
              }
              strokeWidth="2"
            >
              <RNSVGPath
                d="M9.99967 9.97913C9.08301 9.97913 8.33301 9.68746 7.74967 9.10413C7.16634 8.52079 6.87467 7.77079 6.87467 6.85413C6.87467 5.93746 7.16634 5.18746 7.74967 4.60413C8.33301 4.02079 9.08301 3.72913 9.99967 3.72913C10.9163 3.72913 11.6663 4.02079 12.2497 4.60413C12.833 5.18746 13.1247 5.93746 13.1247 6.85413C13.1247 7.77079 12.833 8.52079 12.2497 9.10413C11.6663 9.68746 10.9163 9.97913 9.99967 9.97913ZM3.33301 16.6666V14.7083C3.33301 14.1805 3.46495 13.7291 3.72884 13.3541C3.99273 12.9791 4.33301 12.6944 4.74967 12.5C5.68023 12.0833 6.57259 11.7708 7.42676 11.5625C8.28092 11.3541 9.13856 11.25 9.99967 11.25C10.8608 11.25 11.715 11.3576 12.5622 11.5729C13.4094 11.7882 14.2977 12.0985 15.2272 12.504C15.6619 12.7002 16.0105 12.9847 16.2728 13.3575C16.5352 13.7302 16.6663 14.1805 16.6663 14.7083V16.6666H3.33301ZM4.58301 15.4166H15.4163V14.7083C15.4163 14.4861 15.3504 14.2743 15.2184 14.0729C15.0865 13.8715 14.9233 13.7222 14.7288 13.625C13.84 13.1944 13.0275 12.8993 12.2913 12.7395C11.5552 12.5798 10.7913 12.5 9.99967 12.5C9.20801 12.5 8.43717 12.5798 7.68717 12.7395C6.93717 12.8993 6.12467 13.1944 5.24967 13.625C5.05523 13.7222 4.89551 13.8715 4.77051 14.0729C4.64551 14.2743 4.58301 14.4861 4.58301 14.7083V15.4166ZM9.99967 8.72913C10.5413 8.72913 10.9893 8.55204 11.3434 8.19788C11.6976 7.84371 11.8747 7.39579 11.8747 6.85413C11.8747 6.31246 11.6976 5.86454 11.3434 5.51038C10.9893 5.15621 10.5413 4.97913 9.99967 4.97913C9.45801 4.97913 9.01009 5.15621 8.65592 5.51038C8.30176 5.86454 8.12467 6.31246 8.12467 6.85413C8.12467 7.39579 8.30176 7.84371 8.65592 8.19788C9.01009 8.55204 9.45801 8.72913 9.99967 8.72913Z"
                fill={
                  {
                    "type": 2,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "column",
            "gap": 7,
          }
        }
      >
        <Text
          selectable={true}
          style={
            {
              "color": "hsla(112, 22%, 100%, 1)",
              "fontFamily": "System",
              "fontSize": 36,
              "fontWeight": "700",
              "lineHeight": 19,
            }
          }
          suppressHighlighting={true}
        >
          No Name
        </Text>
        <Text
          selectable={true}
          style={
            {
              "color": "hsla(112, 22%, 59%, 1)",
              "fontFamily": "System",
              "fontSize": 16.8,
              "fontWeight": "400",
              "lineHeight": 19,
              "opacity": 0.6,
            }
          }
          suppressHighlighting={true}
        >
          @
          test
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "display": "none",
          "flexDirection": "row",
          "gap": 24,
        }
      }
    />
  </View>
  <View
    style={
      {
        "borderBottomColor": "hsla(191, 32%, 24%, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "hsla(191, 32%, 24%, 1)",
        "borderLeftWidth": 0,
        "borderRightColor": "hsla(191, 32%, 24%, 1)",
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopColor": "hsla(191, 32%, 24%, 1)",
        "borderTopWidth": 0,
        "flex": 1,
        "flexShrink": 0,
        "height": 0,
        "maxHeight": 0,
        "transform": [
          {
            "translateY": -0.5,
          },
        ],
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "display": "none",
        "flexDirection": "row",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "column",
          "gap": 24,
          "width": 144,
        }
      }
    >
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 59%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "500",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
        Send ID
      </Text>
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 59%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "500",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
        Sendtags
      </Text>
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 59%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "500",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
        Referral Code
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "column",
          "gap": 24,
        }
      }
    >
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 100%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "700",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
        No send id
      </Text>
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 100%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "700",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
         @test
      </Text>
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 100%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "700",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
        <View
          accessibilityLabel="Share"
          accessibilityRole="button"
          cancelable={true}
          disabled={false}
          focusable={[Function]}
          minPressDuration={0}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPress={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          style={
            {
              "color": "hsla(112, 22%, 100%, 1)",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            lineBreakMode="clip"
            numberOfLines={1}
            style={
              {
                "color": "hsla(112, 22%, 100%, 1)",
                "flexGrow": 0,
                "flexShrink": 1,
                "fontFamily": "System",
                "fontSize": 16.8,
                "fontWeight": "400",
                "lineHeight": 19,
              }
            }
            suppressHighlighting={true}
            userSelect="none"
          />
          <View
            pointerEvents="none"
            style={
              {
                "height": 0,
                "minHeight": 0,
                "minWidth": 11.088000000000001,
                "width": 11.088000000000001,
              }
            }
          />
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="20"
            bbWidth="20"
            color="hsla(125, 96%, 40%, 1)"
            fill="none"
            focusable={false}
            height={20}
            meetOrSlice={0}
            minX={0}
            minY={0}
            strokeWidth="2"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "color": "hsla(125, 96%, 40%, 1)",
                },
                {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            tintColor="hsla(125, 96%, 40%, 1)"
            vbHeight={24}
            vbWidth={24}
            width={20}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "strokeWidth",
                ]
              }
              strokeWidth="2"
            >
              <RNSVGPath
                d="M4.19995 10.8V18.48C4.19995 18.9892 4.40224 19.4776 4.76231 19.8376C5.12238 20.1977 5.61074 20.4 6.11995 20.4H17.64C18.1492 20.4 18.6375 20.1977 18.9976 19.8376C19.3577 19.4776 19.56 18.9892 19.56 18.48V10.8M15.72 7.43998L11.88 3.59998M11.88 3.59998L8.03995 7.43998M11.88 3.59998V13.68"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "type": 2,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flexDirection": "column",
        "gap": 32,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "column",
          "gap": 7,
        }
      }
    >
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 59%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "500",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
        Send ID
      </Text>
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 100%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "500",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
        No send id
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "column",
          "gap": 7,
        }
      }
    >
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 59%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "500",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
        Sendtags
      </Text>
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 100%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "500",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
         @test
      </Text>
    </View>
    <View
      style={
        {
          "flexDirection": "column",
          "gap": 7,
        }
      }
    >
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 59%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "500",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
        Referral Code
      </Text>
      <Text
        selectable={true}
        style={
          {
            "color": "hsla(112, 22%, 100%, 1)",
            "fontFamily": "System",
            "fontSize": 19.2,
            "fontWeight": "500",
            "lineHeight": 19,
          }
        }
        suppressHighlighting={true}
      >
        <View
          accessibilityLabel="Share"
          accessibilityRole="button"
          cancelable={true}
          disabled={false}
          focusable={[Function]}
          minPressDuration={0}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onPress={[Function]}
          onPressIn={[Function]}
          onPressOut={[Function]}
          style={
            {
              "color": "hsla(112, 22%, 100%, 1)",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            lineBreakMode="clip"
            numberOfLines={1}
            style={
              {
                "color": "hsla(112, 22%, 100%, 1)",
                "flexGrow": 0,
                "flexShrink": 1,
                "fontFamily": "System",
                "fontSize": 16.8,
                "fontWeight": "400",
                "lineHeight": 19,
              }
            }
            suppressHighlighting={true}
            userSelect="none"
          />
          <View
            pointerEvents="none"
            style={
              {
                "height": 0,
                "minHeight": 0,
                "minWidth": 11.088000000000001,
                "width": 11.088000000000001,
              }
            }
          />
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="20"
            bbWidth="20"
            color="hsla(125, 96%, 40%, 1)"
            fill="none"
            focusable={false}
            height={20}
            meetOrSlice={0}
            minX={0}
            minY={0}
            strokeWidth="2"
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "color": "hsla(125, 96%, 40%, 1)",
                },
                {
                  "flex": 0,
                  "height": 20,
                  "width": 20,
                },
              ]
            }
            tintColor="hsla(125, 96%, 40%, 1)"
            vbHeight={24}
            vbWidth={24}
            width={20}
          >
            <RNSVGGroup
              fill={null}
              propList={
                [
                  "fill",
                  "strokeWidth",
                ]
              }
              strokeWidth="2"
            >
              <RNSVGPath
                d="M4.19995 10.8V18.48C4.19995 18.9892 4.40224 19.4776 4.76231 19.8376C5.12238 20.1977 5.61074 20.4 6.11995 20.4H17.64C18.1492 20.4 18.6375 20.1977 18.9976 19.8376C19.3577 19.4776 19.56 18.9892 19.56 18.48V10.8M15.72 7.43998L11.88 3.59998M11.88 3.59998L8.03995 7.43998M11.88 3.59998V13.68"
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "stroke",
                    "strokeWidth",
                    "strokeLinecap",
                    "strokeLinejoin",
                  ]
                }
                stroke={
                  {
                    "type": 2,
                  }
                }
                strokeLinecap={1}
                strokeLinejoin={1}
                strokeWidth="2"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "borderBottomColor": "hsla(191, 32%, 24%, 1)",
        "borderBottomWidth": 1,
        "borderLeftColor": "hsla(191, 32%, 24%, 1)",
        "borderLeftWidth": 0,
        "borderRightColor": "hsla(191, 32%, 24%, 1)",
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopColor": "hsla(191, 32%, 24%, 1)",
        "borderTopWidth": 0,
        "flex": 1,
        "flexShrink": 0,
        "height": 0,
        "maxHeight": 0,
        "transform": [
          {
            "translateY": -0.5,
          },
        ],
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "column",
        "gap": 24,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "gap": 24,
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;
